src/cli/runtime.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * Runtime detection utilities for srvx CLI
+ */
+
+export type SupportedRuntime = "node" | "deno" | "bun";
+
+/**
+ * Detect the current JavaScript runtime
+ */
+export function detectRuntime(): SupportedRuntime {
+  // Check for Deno
+  if (typeof Deno !== "undefined") {
+    return "deno";
+  }
+
+  // Check for Bun
+  if (typeof Bun !== "undefined") {
+    return "bun";
+  }
+
+  // Default to Node.js
+  return "node";
+}
+
+/**
+ * Get the appropriate srvx adapter import path for the detected runtime
+ */
+export function getAdapterImport(runtime: SupportedRuntime): string {
```

Comment:
We do not need this. Simply importing `srvx` in each runtime, loads serve function for it based on export contidions.